src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasResultSourceService.cs

```diff
@@ -0,0 +1,384 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.Caching;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+using CSharpFunctionalExtensions;
+
+using EnvDTE;
+
+using EnvDTE80;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.Sarif.Viewer;
+using Microsoft.Sarif.Viewer.ResultSources.Domain;
+using Microsoft.Sarif.Viewer.ResultSources.Domain.Abstractions;
+using Microsoft.Sarif.Viewer.ResultSources.Domain.Models;
+using Microsoft.Sarif.Viewer.ResultSources.Domain.Services;
+using Microsoft.Sarif.Viewer.Shell;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+
+using Newtonsoft.Json;
+
+using Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core.Models;
+
+using Result = CSharpFunctionalExtensions.Result;
+
+namespace Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core.Services
+{
+    public class DevCanvasResultSourceService : IResultSourceService
+    {
+        public int FirstMenuId { get; set; }
```

Comment:
What are 'FirstMenuId' and 'FirstCommandId'

Reply:
Not super familiar with it, I know it's used in the context menu of the error list but not super sure what that means either. Got the values from Chris

Reply:
The result source platform reserves a set of menu IDs on the Error List context menu for each extension. The way VS handles this is declarative, which makes even basic dynamic-ness difficult. (I know it's how Win32 has always done menus, but there has to be SOME way to modernize it!)